pkg/cloud/client.go

```diff
@@ -66,6 +71,12 @@ type SecretConfig struct {
 	StringData Config            `yaml:"stringData"`
 }
 
+var clientCache *ttlcache.Cache
+var cacheMutex sync.Mutex
+var log = ctrl.Log.WithName("Client")
+
+const cacheExpiration = time.Duration(1 * time.Minute)
```

Comment:
Please make this externally configurable.

Reply:
Also, I'd vote for a much longer TTL.  In the previous incarnation connections lived indefinitely.  How about 1 hour?

Reply:
I will change the expiration to 1 hour. How do you want it to be configurable externally? Via an environment variable?

Reply:
This may be a can of worms.  But one day we'll want to change this in production and won't want to code/release to do so.

Env var isn't great, but better than nothing.  The old cloud-config had a global section which could have accommodated that, but the new CAPC secrets-based identities are all FD-specific.  Maybe we have a need for a global config secret (or config map).